File: CS/VGridControl/CustomVGridControl/BoundDataModeHelperDescendant.cs
using DevExpress.XtraVerticalGrid.Data;

namespace E4423
{
    public class CustomDataModeHelper : BoundDataModeHelper
    {
        public CustomDataModeHelper(VGridDataManager manager)
            : base(manager)
        {

        }

        public override bool NewItemRecordMode
        {
            get
            {
                if ((this.Manager.CurrentClient as VGridControlDescendant).AllowNew)
                    return true;
                return false;
            }
        }
       
        public override int Position
        {
            get
            {
                if(Manager.CurrentControllerRow == -2147483648) 
                    return RecordCount - 1;
				return Manager.CurrentControllerRow;
            }
        }
    }
}

File: VB/VGridControl/CustomVGridControl/BoundDataModeHelperDescendant.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraVerticalGrid.Data

Namespace E4423
	Public Class CustomDataModeHelper
		Inherits BoundDataModeHelper
		Public Sub New(ByVal manager As VGridDataManager)
			MyBase.New(manager)

		End Sub

		Public Overrides ReadOnly Property NewItemRecordMode() As Boolean
			Get
				If (TryCast(Me.Manager.CurrentClient, VGridControlDescendant)).AllowNew Then
					Return True
				End If
				Return False
			End Get
		End Property

		Public Overrides ReadOnly Property Position() As Integer
			Get
				If Manager.CurrentControllerRow = -2147483648 Then
					Return RecordCount - 1
				End If
				Return Manager.CurrentControllerRow
			End Get
		End Property
	End Class
End Namespace
